activestorage/lib/active_storage/attached/many.rb

```diff
@@ -3,7 +3,7 @@
 module ActiveStorage
   # Decorated proxy object representing of multiple attachments to a model.
   class Attached::Many < Attached
-    delegate_missing_to :attachments
+    delegate_missing_to :attachments, allow_nil: true
```

Comment:
`:allow_nil` isn’t necessary in this case. `attachments` will always return an `Array` or `ActiveRecord::Associations::CollectionProxy`.

Reply:
Hi George I hope you are fine!

I am building an App where I have two kinds of users (its a platform) and in one of this cases the Avatar must be compulsory in other hand for the end-user it is optional. Given the fact that I am tried to minimize the numbers of fields in the form to avoid registration abandon.

**Important**: I am use has_one_attached 
After read the discussion above I assume that  the behavior "  delegate_missing_to :attachment, allow_nil: true" is already in place when I ran the command "rails active_storage:install". Is that true?

My actual behavior:
When I try to create a user without avatar, it is created, but the active recorde returns an error:

  User Create (0.5ms)  INSERT INTO "users" ("encrypted_password", "created_at", "updated_at", "cpf", "full_name", "phone_number", "is_company", "status", "country", "authentication_token") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) RETURNING "id"  [["encrypted_password", "$2a$11$dmNXTW1etZKXqpLhvhT0c.7FFv5V4u3TffsyLgInIHKFtMO4zd5K6"], ["created_at", "2020-02-10 23:13:06.886745"], ["updated_at", "2020-02-10 23:13:06.886745"], ["cpf", "62056594099"], ["full_name", "Silvia Brandao"], ["phone_number", "51998909909"], ["is_company", false], ["status", "Ativo"], ["country", "BR"], ["authentication_token", "v-HXKQxPrTPSxoGBoApX"]]
  ↳ app/controllers/users/registrations_controller.rb:15:in `create'
   (1.1ms)  COMMIT
  ↳ app/controllers/users/registrations_controller.rb:15:in `create'
  User Load (0.9ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 8], ["LIMIT", 1]]
  ↳ app/controllers/users/registrations_controller.rb:15:in `create'
  ActiveStorage::Attachment Load (1.7ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 9], ["record_type", "User"], ["name", "photo"], ["LIMIT", 1]]
  ↳ app/controllers/users/registrations_controller.rb:15:in `create'
Completed 500 Internal Server Error in 332ms (ActiveRecord: 12.1ms | Allocations: 33457)

Db:shema:
create_table "users", force: :cascade do |t|
    t.string "email", default: ""
    t.string "encrypted_password", default: "", null: false
    t.string "reset_password_token"
    t.datetime "reset_password_sent_at"
    t.datetime "remember_created_at"
    t.datetime "created_at", precision: 6, null: false
    t.datetime "updated_at", precision: 6, null: false
    t.string "cpf"
    t.string "cnpj"
    t.string "full_name"    
    t.integer "photo", array: true
end

I would realy appreciate your help!